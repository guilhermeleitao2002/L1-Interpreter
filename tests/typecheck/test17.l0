type Node = struct { 
  #value: int, 
  #next: ref<union { #none:(), #some:Node }> 
};

let n1:Node = { #value=1, #next=box(#none(())) };
let n2:Node = { #value=2, #next=box(#some(n1)) };
n1.#next := #some(n2);

match !(n2.#next) {
  #none(_) -> 0
  | #some(node) -> node.#value
};;