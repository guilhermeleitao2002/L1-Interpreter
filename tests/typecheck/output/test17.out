TypeError: Expression type struct { next: ref<union { none: () }>, value: int } does not match declared type struct { next: ref<union { some: Node, none: () }>, value: int }